### PR TITLE
[INLONG-5020][Manager] Refactoring InnerInlongManagerClient of manager

### LINKED ISSUE #5020: [Improve][Manager] Refactoring InnerInlongManagerClient of manager client
### Description

Refactoring `InnerInlongManagerClient` of manager client , 
because it will get bigger so that it becomes less readable.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by haifxu**
### [INLONG-5020][Manager] Refactoring InnerInlongManagerClient of manager

- Fixes #5020 

### Modifications

Split `InnerInlongManagerClient` into multiple clients to prevent it from being too large.

### Verifying this change

- [x] This change is already covered by existing tests, such as:
- org/apache/inlong/manager/client/api/impl/InlongGroupImplTest.java
- org/apache/inlong/manager/client/api/impl/InlongStreamImplTest.java
- org/apache/inlong/manager/client/api/inner/InnerInlongManagerClientTest.java

### CODE REVIEW

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/WorkflowClient.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+        return response.getData();
+    }
+
+    /**
+     * get inlong group error messages
+     */
+    public List<EventLogView> getInlongGroupError(String inlongGroupId) {
+        Response<PageInfo<EventLogView>> response = ClientUtils.executeHttpCall(
+                workflowApi.getInlongGroupError(inlongGroupId, -1));
```
Comment (vernedeng) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/WorkflowClient.java:80]: magic number -1 , plz add comment to explan what "-1" stand for

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/WorkflowClient.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        inlongGroupApproveForm.putPOJO("groupApproveInfo", newGroupProcessForm.getGroupInfo());
+        inlongGroupApproveForm.putPOJO("streamApproveInfoList", newGroupProcessForm.getStreamInfoList());
+        inlongGroupApproveForm.put("formName", "InlongGroupApproveForm");
+        workflowTaskOperation.set("form", inlongGroupApproveForm);
+
+        log.info("startInlongGroup workflowTaskOperation: {}", inlongGroupApproveForm);
+
+        Map<String, Object> requestMap = JsonUtils.OBJECT_MAPPER.convertValue(workflowTaskOperation,
+                new TypeReference<Map<String, Object>>() {
+                });
```
Comment (vernedeng) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/WorkflowClient.java:67]: no need to make a new line

File: inlong-manager/manager-client-tools/src/main/java/org/apache/inlong/manager/client/cli/util/ClientUtils.java [hunk 3]
```diff
@@ -84,4 +85,17 @@ public static String readFile(File file) {
         return null;
     }
 
+    private static void initClientConfiguration() throws IOException {
+        Properties properties = new Properties();
+        String path = Thread.currentThread().getContextClassLoader().getResource("").getPath() + CONFIG_FILE;
+        InputStream inputStream = new BufferedInputStream(Files.newInputStream(Paths.get(path)));
```
Comment (leosanqing) [inline inlong-manager/manager-client-tools/src/main/java/org/apache/inlong/manager/client/cli/util/ClientUtils.java:91]: please use try-with-resource to close the stream

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/InlongGroupImpl.java [hunk 4]
```diff
@@ -62,30 +67,38 @@ public class InlongGroupImpl implements InlongGroup {
...
+        this.streamClient = clientFactory.getStreamClient();
+        this.groupClient = clientFactory.getGroupClient();
+        this.workFlowClient = clientFactory.getWorkflowClient();
+
+//        this.groupClient =
 
-        InlongGroupInfo newGroupInfo = managerClient.getGroupIfExists(groupInfo.getInlongGroupId());
+        InlongGroupInfo newGroupInfo = groupClient.getGroupIfExists(groupInfo.getInlongGroupId());
         if (newGroupInfo != null) {
             this.groupContext.setGroupInfo(groupInfo);
```
Comment (leosanqing) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/InlongGroupImpl.java:90]: Should this be newGroupInfo?

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/WorkflowClient.java (hunk 2 above)
Comment (haifxu) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/WorkflowClient.java:66]: This is the code style for anonymous inner classes, so it must be in a new line.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/WorkflowClient.java (hunk 1 above)
Comment (haifxu) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/WorkflowClient.java:79]: This method is just copied from innerClient, but I'll figure out what it means and comment.

File: inlong-manager/manager-client-tools/src/main/java/org/apache/inlong/manager/client/cli/util/ClientUtils.java (hunk 3 above)
Comment (haifxu) [reply inlong-manager/manager-client-tools/src/main/java/org/apache/inlong/manager/client/cli/util/ClientUtils.java]: done

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/ClientUtils.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+ */
+@Slf4j
+@UtilityClass
+public class ClientUtils {
+
+    private static final String REQUEST_FAILED_MSG = "Request to Inlong %s failed: %s";
+
+    private static ClientFactory clientFactory;
+
+    public static ClientFactory getClientFactory(ClientConfiguration configuration) {
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/ClientUtils.java:50]: Please add Java docs for those public static methods.

**Review by healchow — CHANGES_REQUESTED** [review]

**Review by kipshi — APPROVED** [review]

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/ClientUtils.java (hunk 5 above)
Comment (haifxu) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/ClientUtils.java:56]: Done.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/InlongStreamImpl.java [hunk 6]
```diff
@@ -67,10 +82,17 @@ public class InlongStreamImpl implements InlongStream {
     /**
      * Constructor of InlongStreamImpl.
      */
-    public InlongStreamImpl(InlongStreamInfo streamInfo, InnerInlongManagerClient managerClient) {
-        this.managerClient = managerClient;
+    public InlongStreamImpl(InlongStreamInfo streamInfo, ClientConfiguration configuration) {
+//        this.managerClient = managerClient;
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/InlongStreamImpl.java:86]: Why comment on this line? If it is unnecessary, please remove it.

Comment (haifxu) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/InlongStreamImpl.java]: Removed.

**Review by healchow — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]
lgtm

**Review by leosanqing — APPROVED** [review]

**Pull request merged**